crates/circuit/src/operations.rs

```diff
@@ -107,6 +107,67 @@ impl Param {
         }
     }
 
+    /// Add two [Param]s.
+    ///
+    /// This does not support additions involving a [Param::Obj] variant.
+    pub fn add(&self, other: &Param) -> Param {
+        match (&self, other) {
+            (_, Param::Float(right)) => self.add_f64(*right),
+            (Param::Float(left), _) => other.add_f64(*left), // add is commutative here
+            (Param::ParameterExpression(left), Param::ParameterExpression(right)) => {
+                // TODO we could properly propagate the error here
+                Param::ParameterExpression(Arc::new(
+                    left.add(right).expect("Name conflict during add."),
+                ))
+            }
+            _ => unreachable!("Unsupported addition."),
```

Comment:
This is not unreachable. Unreachable isn't for marking "I don't want you to get here", it's for marking "this is literally impossible but the compiler can't reason about it".

Reply:
A nicer way would be to have a proper error here -- we should also propagate the name-conflict error that we currently just `.expect()`

Reply:
The panic is suitable if it could only have been caused by an internal logic error within the Rust code, but yeah, a non-aborting error for the name conflict would be nicer.